Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcContentServer.cs

```diff
@@ -284,6 +287,82 @@ private async Task CopyFileAsync(CopyFileRequest request, IServerStreamWriter<Co
             }
         }
 
+        /// <summary>
+        /// Implements a request copy file request
+        /// </summary>
+        private async Task<RequestCopyFileResponse> RequestCopyFileAsync(RequestCopyFileRequest request, CancellationToken cancellationToken)
+        {
+            OperationStarted();
+
+            var startTime = DateTime.UtcNow;
+            var cacheContext = new OperationContext(new Context(new Guid(request.TraceId), _logger), cancellationToken);
+
+            var sessionResult = await GetRequestCopySessionAsync(cacheContext, request.CacheName);

```

Comment:
>GetRequestCopySessionAsync [](start = 38, length = 26)

I don't think this should be on the session level. I think it should be IFileCopyingStore instead. Doesn't make sense to create a session just to initiate a copy.

Reply:
Is it possible to do a Put without a session? Because technically this is not just a copy, its a copy and then a put

Reply:
Is it possible to do a Put without a session? Because technically this is not just a copy, its a copy and then a put

---
In reply to: [311172395](https://github.com/microsoft/BuildXL/pull/705#discussion_r311172395) [](ancestors = 311172395)

Reply:
Indeed this would involve a non-trivial amount of shuffling. That said, we still need to shutdown the session.

---
In reply to: [312661540](https://github.com/microsoft/BuildXL/pull/705#discussion_r312661540) [](ancestors = 312661540,311172395)

Reply:
Decided to complete PR as is and worry about this at a later time

---
In reply to: [312988309](https://github.com/microsoft/BuildXL/pull/705#discussion_r312988309) [](ancestors = 312988309,312661540,311172395)